[Spike] Add `evm` contents (#12)
futureversecom/trn-docs: docs/native/cookbook/asset.md

---
title: Asset
---

On the Root Network (TRN), the term 'asset' encompasses everything from fungible to non-fungible and semi-fungible tokens. This page outlines common use cases when working with assets on TRN.

:::info

The instructions provided here are kept concise, focusing on the high-level aspects. If you need more in-depth implementation details and runnable examples, checkout [`trn-examples`](https://github.com/futureversecom/trn-examples/tree/main/examples/substrate) repository

:::

> ## How do I create and mint NFT on TRN?

Reviewer: ```suggestion
## How do I create and mint an NFT on TRN?
```

### 1. Create Collection

To create your NFT collection, call the `nft.createCollection()` extrinsic with the following parameters

```typescript
await api.tx.nft.createCollection(
  collectionName,
  initialIssuance,
  maxIssuance,
  tokenOwner,
  metadataScheme,
  royaltiesSchedule,
  crossChainCompatibility
);
```

- `collectionName` - The name of the collection
- `initialIssuance` - Number of tokens to mint now
- `maxIssuance` - Maximum number of tokens allowed in collection

> - `tokenOwner` - The token owner, defaults to the caller
> - `metadataScheme` - The off-chain metadata referencing scheme for tokens in this
> - `royaltiesSchedule` - The defacto royalties plan for secondary sales, this will apply to all tokens in the collection by default. e.g
> - `crossChainCompatibility` - The cross-chain compatibility flag

Reviewer: ```suggestion
- `tokenOwner` - The receiver of the tokens minted now, defaults to the caller
- `metadataScheme` - The off-chain metadata referencing scheme for tokens in this collection
- `royaltiesSchedule` - The defacto royalties plan for secondary sales, this will apply to all tokens in the collection
- `crossChainCompatibility` - The cross-chain compatibility flag allows for limited compatibility with XRPL at the cost of higher mint fees
```

:::caution

> The `royaltiesSchedule` is in permil scaling, so if you want a 10% royalty, that would be `100000` (as `100_000 / 1_000_000 = 0.1`);

Reviewer: ```suggestion
The `royaltiesSchedule` is in permill scaling, so if you want a 10% royalty, that would be `100,000` (as `100_000 / 1_000_000 = 0.1`);
```

:::

### 2. Mint NFT(s)

> Once your collection is created, you can call the `nft.mint()` method on the collection contract to start minting

Reviewer: ```suggestion
Once your collection is created, the collection owner can call the `nft.mint()` method on the collection contract to start minting additional tokens so long as the max issuance has not been reached
```

```typescript
await api.tx.nft.mint(collectionId, quantity, beneficiary);
```

> - `collectionId` - The collection id

Reviewer: ```suggestion
- `collectionId` - The unique collection id assigned when the collection was created
```
